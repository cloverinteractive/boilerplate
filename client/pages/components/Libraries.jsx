// @flow

import React from 'react';
import { v4 } from 'uuid';
import libraries from 'pages/constants/libraries';

type NpmLibrary = {
  description: string,
  library: string,
  url: string,
};

const Library = ({ description, library, url }: NpmLibrary) => (
  <li>
    <a href={url} rel="noopener noreferrer" target="_blank">{library}</a> - <em>{description}</em>
  </li>
);

const Libraries = () => {
  const items = libraries.map(library => <Library key={v4()} {...library} />);

  return (
    <ul className="libraries">
      { items }
    </ul>
  );
};

export default Libraries;
