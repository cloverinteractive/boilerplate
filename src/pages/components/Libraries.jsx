import React from 'react';
import v4 from 'uuid/v4';

export const Library = ({ description, library, url }) => (
  <li>
    <a href={url} rel="noopener noreferrer" target="_blank">{library}</a> - <em>{description}</em>
  </li>
);

const Libraries = ({ libraries }) => {
  const items = libraries.map(library => <Library key={v4()} {...library} />);

  return (
    <ul className="libraries">
      { items }
    </ul>
  );
};

export default Libraries;
