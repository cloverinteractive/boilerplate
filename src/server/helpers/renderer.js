import { renderToString } from 'react-dom/server';
import { Helmet } from 'react-helmet';
import serialize from 'serialize-javascript';
import { isDevelopment } from 'server/helpers/env-access';

export default (Component, store) => {
  const content = renderToString(Component);
  const helmet = Helmet.renderStatic();

  const productionStyles =
    `<link rel="stylesheet" href="/vendors.css">
    <link rel="stylesheet" href="/styles.css">`;

  const styles = isDevelopment ? '' : productionStyles;

  const template =
    `<!doctype html>
    <html lang="en-US">
      <head>
        <meta charset="utf-8">
        ${helmet.link.toString()}
        ${helmet.title.toString()}
        ${helmet.meta.toString()}
        ${styles}
        <script src="/vendors.js"></script>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())};
        </script>
        <script async src="/bundle.js"></script>
      </body>
    </html>`;

  return template;
};
