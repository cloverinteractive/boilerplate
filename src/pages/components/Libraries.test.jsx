import * as React from 'react';
import renderer from 'react-test-renderer';
import Libraries from './Libraries';

jest.mock('uuid/v1', () => jest.fn((_, increment) => `mocked-random-uuid-${increment}`));

describe('<Libraries />', () => {
  const libraries = [
    { description: 'A library description', library: 'A Library', url: 'https://alibrary.com' },
    { description: 'B library description', library: 'B Library', url: 'https://blibrary.com' },
  ];

  test('it renders correctly', () => {
    const tree = renderer
      .create(<Libraries libraries={libraries} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
