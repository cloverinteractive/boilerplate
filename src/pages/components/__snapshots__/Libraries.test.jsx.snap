// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Libraries /> it renders correctly 1`] = `
<ul
  className="libraries"
>
  <li>
    <a
      href="https://alibrary.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      A Library
    </a>
     
    -
    <span
      className="is-italic"
    >
      A library description
    </span>
  </li>
  <li>
    <a
      href="https://blibrary.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      B Library
    </a>
     
    -
    <span
      className="is-italic"
    >
      B library description
    </span>
  </li>
</ul>
`;
